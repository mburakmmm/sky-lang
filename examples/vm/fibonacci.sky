# Fibonacci - VM Recursion Test
# Demonstrates bytecode VM's ability to handle deep recursion

function fib(n: int): int
  if n <= 1
    return n
  end
  let a = fib(n - 1)
  let b = fib(n - 2)
  return a + b
end

function main
  print("Fibonacci with VM (No Stack Overflow!)")
  print(fib(10))
  print(fib(15))
  print(fib(20))
  print(fib(25))
  print(fib(30))
  print("Success!")
end

