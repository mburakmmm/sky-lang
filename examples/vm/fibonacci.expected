Fibonacci with VM (No Stack Overflow!)
55
610
6765
75025
832040
Success!
